transactions/transactions_service.go

```diff
@@ -152,6 +162,16 @@ func (svc *transactionsService) MakeInvoice(ctx context.Context, amount uint64,
 		}
 	}
 
+	if metadata["app_id"] != nil {
+		overwriteAppIdType, ok := metadata["app_id"].(float64)
+		if !ok {
+			return nil, errors.New("failed to overwrite app ID")
+		}
+		overwriteAppId := uint(overwriteAppIdType)
```

Comment:
do we need to validate that this app exists? 
what if an invalid id is set? (or the app is deleted by now)

Reply:
I tested this case and I think it's fine without further validation. It will just be treated as a normal invoice not for a sub-wallet. (Kind of similar if you delete an app, the app will be deleted but transactions will remain).

A very small side effect I found: if you create an invoice for an app ID that doesn't exist yet (e.g. I had 2 local apps but made one for id:3) when the app is created, the app will suddenly have that transaction associated with it.